### PR TITLE
update pangea3 host-config file and remove hard-coded Python version 

### PR DESCRIPTION
**Pull request by XL64**

### CODE REVIEW

File: host-configs/TOTAL/pangea3-gcc8.4.1-openmpi-4.1.2.cmake [hunk 1]
```diff
@@ -44,7 +43,7 @@ if (DEFINED ENV{MPI_ROOT})
   set(MPIEXEC                $ENV{MPI_ROOT}/bin/mpirun  CACHE STRING "")
   set(ENABLE_WRAP_ALL_TESTS_WITH_MPIEXEC ON CACHE BOOL "")
 else()
-  message(FATAL_ERROR "You must have MPI_ROOT variable set, we advise loading module smpi/10.3.0.1")
+  message(FATAL_ERROR "You must have MPI_ROOT variable set, we advise loading module openmpi/p3/4.1.2")
```
Comment (TotoGaz) [inline host-configs/TOTAL/pangea3-gcc8.4.1-openmpi-4.1.2.cmake:46]: If you want a specific version, maybe can you check that `MPI_ROOT` has the exact proper value? (up to you, of course.)

File: src/pygeosx/CMakeLists.txt [hunk 2]
```diff
@@ -21,11 +21,11 @@ add_custom_command( TARGET pygeosx
 
 add_custom_command( TARGET pygeosx
                     POST_BUILD
-                    COMMAND ln -sf ${CMAKE_BINARY_DIR}/lib/pygeosx.so ${CMAKE_BINARY_DIR}/lib/PYGEOSX/lib/python3.8/site-packages/ )
+                    COMMAND ln -sf ${CMAKE_BINARY_DIR}/lib/pygeosx.so ${CMAKE_BINARY_DIR}/lib/PYGEOSX/lib/python${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR}/site-packages/ )
```
Comment (TotoGaz) [inline src/pygeosx/CMakeLists.txt:24]: @cssherman Does that sound good to you?
I tend to be OK but I don't really know what's going on there.

**Review by TotoGaz — COMMENTED** [review]
LGTM

File: src/pygeosx/CMakeLists.txt (hunk 2 above)
Comment (cssherman) [reply src/pygeosx/CMakeLists.txt:24]: Looks great - I had to do the same in a test branch myself recently

**Comment by TotoGaz** [comment]
@XL64 Do you want to go in the merge queue right now?
Or do you want to be in sync with `TPL`?

**Comment by XL64** [comment]
> @XL64 Do you want to go in the merge queue right now? Or do you want to be in sync with `TPL`?

I don't need it to be synced.

File: host-configs/TOTAL/pangea3-gcc8.4.1-openmpi-4.1.2.cmake (hunk 1 above)
Comment (XL64) [reply host-configs/TOTAL/pangea3-gcc8.4.1-openmpi-4.1.2.cmake:46]: That message is not correct this module does not exist anymore i think, i'll have to check. Please wait before merge.

**Comment by TotoGaz** [comment]
@cssherman OK for the pygeosx part?

**Review by cssherman — APPROVED** [review]

**Pull request merged**